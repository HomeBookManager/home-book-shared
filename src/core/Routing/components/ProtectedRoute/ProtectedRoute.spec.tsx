// import { Provider } from 'react-redux';
// import { BrowserRouter as Router, Redirect } from 'react-router-dom';
// import { render } from '@testing-library/react';

// // components
// import ProtectedRoute from './ProtectedRoute';

// // mocks
// import { routerStateMock } from '../../../../tests/mocks/reducer/routerMock';

// // others
// import { TAppRouteData } from '../../types';

// // store
// import { configureStore } from '../../../../store/store';

// // types
// import { RouteName } from '../../constants/routes';

// // utils
// import { history } from '../../../../utils/history';

// const stateMock = {
//   ...routerStateMock,
// };

// const routeData: TAppRouteData = {
//   Component: () => <div>Component</div>,
//   guards: [
//     {
//       guardCheck: () => true,
//       renderFallback: () => <Redirect exact from={'/'} to={'/error'} />,
//       translationKey: '',
//     },
//   ],
//   name: RouteName.test,
// };

// describe('ProtectedRoute snapshots', () => {
//   // mock
//   const store = configureStore(stateMock);

//   it('should allow user on route', () => {
//     // before
//     const { Component, guards, name } = routeData;
//     const { asFragment } = render(
//       <Provider store={store}>
//         <Router>
//           <ProtectedRoute component={Component} guards={guards} name={name} />
//         </Router>
//       </Provider>,
//     );

//     // result
//     expect(asFragment()).toMatchSnapshot();
//   });

//   it('should allow rdner without guards', () => {
//     // before
//     const { Component, name } = routeData;
//     const { asFragment } = render(
//       <Provider store={store}>
//         <Router>
//           <ProtectedRoute component={Component} name={name} />
//         </Router>
//       </Provider>,
//     );

//     // result
//     expect(asFragment()).toMatchSnapshot();
//   });

//   it('should not allow user on route', () => {
//     // mock
//     const store = configureStore(stateMock);

//     // before
//     const { Component, guards, name } = routeData;
//     const { asFragment } = render(
//       <Provider store={store}>
//         <Router>
//           <ProtectedRoute
//             component={Component}
//             guards={[{ ...guards![0], guardCheck: () => false }]}
//             name={name}
//           />
//         </Router>
//       </Provider>,
//     );

//     // result
//     expect(asFragment()).toMatchSnapshot();
//   });
// });
