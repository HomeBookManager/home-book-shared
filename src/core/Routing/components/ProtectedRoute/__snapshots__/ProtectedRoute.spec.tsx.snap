// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoute snapshots should allow rdner without guards 1`] = `<DocumentFragment />`;

exports[`ProtectedRoute snapshots should allow user on route 1`] = `<DocumentFragment />`;

exports[`ProtectedRoute snapshots should not allow user on route 1`] = `<DocumentFragment />`;
